airflow/configuration.py

```diff
@@ -469,11 +469,22 @@ class AirflowConfigParser(ConfigParser):
     }
 
     def __init__(self, *args, **kwargs):
+        default_config = kwargs.pop('default_config', None)
         ConfigParser.__init__(self, *args, **kwargs)
-        self.read_string(parameterized_config(DEFAULT_CONFIG))
+        self.defaults = ConfigParser(*args, **kwargs)
+
+        AirflowConfigParser._read_string(
+            self.defaults, parameterized_config(DEFAULT_CONFIG))
+        if default_config:
+            AirflowConfigParser._read_string(
+                self.defaults, parameterized_config(default_config))
         self.is_validated = False
 
     def read_string(self, string, source='<string>'):
+        AirflowConfigParser._read_string(self, string, source)
+
+    @staticmethod
```

Comment:
It requires unicode string when it reads config from a string in the backported library. So the config string has to be explicitly converted into unicode before passing the parameter.

For this, I think monkey patching can be better option.